src/tenderly/deploy-verify-zns.ts

```diff
@@ -0,0 +1,50 @@
+import { deployZNS } from "../../test/helpers";
+import * as hre from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+
+export const deployVerifyZNS = async ({
+  governor,
+  governorAddresses = [ governor.address ],
+  adminAddresses = [ governor.address ],
+} : {
+  governor : SignerWithAddress;
+  governorAddresses ?: Array<string>;
+  adminAddresses ?: Array<string>;
+}) => {
+  const zns = await deployZNS({
+    deployer: governor,
+    governorAddresses,
+    adminAddresses,
+  });
+
+  // TODO: find a way to get these names from the object
+  const names = [
+    "ZNSAccessController",
+    "ZNSRegistry",
+    "ZNSDomainToken",
+    "ZeroTokenMock",
+    "ZNSAddressResolver",
+    "ZNSPriceOracle",
+    "ZNSTreasury",
+    "ZNSEthRegistrar",
+  ];
+
+  // Verify and add source code to Tenderly
+  // for easy debugging
+  await Object.values(zns).reduce(
```

Comment:
`forEach` may not work with promised properly as some other looper functions.
the reason I'm using `reduce` here is because it can properly await promises and correct times.
I also really like `reduce` because it can be used in whichever way you want and it works a bit differently than other array methods. with async code iteration through reduce is my favorite because it's predictable.